pymeasure/instruments/scientificmagnetics/scimag.py

```diff
@@ -0,0 +1,448 @@
+
+
+class SciMag(Instrument):
+    """ Represents the Scientific Magnetics (Twickenham Scientific Instruments)
+    Superconducting Magnet Controller SciMag S-11-52-13.
+
+    Provides a high-level interface for
+    interacting with the instrument.
+    WARNING: Use at your own risk!
+    Faulty operation of a superconducting magnet can lead to severe damage or injury!
+    Read the manual: https://www.twicksci.co.uk/manuals/pdf/smc552+.pdf
+    """
+
+    _RAMP_RATE_MAX = 5
+    _VOLTAGE_LIMIT_MAX = 1
+
+    pause = Instrument.setting(
+        "P%d",
+        """A boolean property that controls the pause-status,
+         which takes the values True (pause) and False (continue)
+         """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    unit = Instrument.setting(
+        "T%d",
+        """Sets the units displayed and can take the values
+         "A"mpere or "T"esla
+         """,
+        validator=strict_discrete_set,
+        values={"A": 0, "T": 1},
+        map_values=True
+    )
+
+    reverse_switch_on = Instrument.setting(
+        "D%d",
+        """Sets the direction of the reversing switch, if installed.
+        Forward if switch is off; Reverse if switch is on.
+        Takes boolean value. True = Switch is on, Field is reverse.
+        WARNING: Change only if Magnet has been ramped to Zero!
+        """,
+        validator=strict_discrete_set,
+        values={False: 0, True: 1},
+        map_values=True
+    )
+
+    ramp_target = Instrument.setting(
+        "R%d",
+        """Sets Ramp Target. Can take the values "Z"ero, "L"ower, "U"pper.
+        """,
+        validator=strict_discrete_set,
+        values={"Z": 0, "L": 1, "U": 2},
+        map_values=True
+    )
+
+    persistent_mode_heater = Instrument.setting(
+        "H%d",
+        """Sets the persistent mode.
+        Can take the values "O"ff, "C"onditional on, "U"nconditional on, "R"eset persistent current mode to zero.
+        WARNING: Unconditional on will turn on the heater without any interlocks! Use at own risk.
+        Standard mode to switch heater on is Conditional.
+        Simplified Commands are "True" (Conditional on) and "False" (Off).
+        """,
+        validator=strict_discrete_set,
+        values={"O": 0, "C": 1, "U": 2, "R": 9, False: 0, True: 1},
+        map_values=True
+    )
+
+    ramp_rate = Instrument.setting(
+        "A%08.5f",
+        """Sets the Ramp rate in A/s.""",
+        validator=strict_range,
+        values=[0, _RAMP_RATE_MAX],
+    )
+
+    external_trip = Instrument.setting(
+        "X%d",
+        """Sets the External trip:
+         False = Off, True = Simple trip On, "Auto" = Auto-rampdown On.""",
+        validator=strict_discrete_set,
+        values={False: 0, True: 1, "Auto": 4},
+        map_values=True
+    )
+
+    terminal_voltage_limit = Instrument.setting(
+        "Y%04.1f",
+        """Sets the terminal voltage limit to nn.n V.""",
```

Comment:
what do you mean by `nn.n`? If you mean that you can set it with 0.1V accuracy, it might be clearer to directly mention that.